### PR TITLE
email_template_qweb: ease override of values, include defaults

### PR DESCRIPTION
**Pull request by simahawk**
Improvements:

1. provide hook to override rendering values
2. provide default rendering values as per Jinja templates

### CODE REVIEW

File: email_template_qweb/models/mail_template.py [hunk 1]
```diff
@@ -40,3 +42,36 @@ def generate_email(self, res_ids, fields=None):
...
+    def _qweb_render_context(self, record):
+        res = {}
+        if record._name == 'mail.message':
+            partner_model = self.env['res.partner']
+            # these variables are usually loaded when the notification is sent
+            # but there are some key values that that are nice to have
+            # if you want to pre-render your template in preview mode.
+            # This is particularly useful if you use the template
+            # from the mail composer, whereas the rendering is done on load.
+            # Also, you don't need to access them from `ctx`.
```
Comment (hbrunn) [inline email_template_qweb/models/mail_template.py]: I believe all of this should be the docstring of this function, with a little bit of reformulation.

File: email_template_qweb/models/mail_template.py [hunk 2]
```diff
@@ -40,3 +42,36 @@ def generate_email(self, res_ids, fields=None):
...
+            res = partner_model._notify_prepare_template_context(record)
+        res.update({
+            'object': record,
+            'email_template': self,
+            # Same as for Jinja rendering,
+            # taken from `mail_template.render_template`.
+            # These ease porting of old Jinja templates to qweb ones.
+            'format_date':
+                lambda date, format=False,
+                context=self._context: format_date(self.env, date, format),
```
Comment (hbrunn) [inline email_template_qweb/models/mail_template.py:83]: I don't see why we need jinja compatibility. If you want to format a date or money, you just use the proper widget in qweb

**Review by hbrunn — CHANGES_REQUESTED** [review]
please give an argument why not to use the qweb equivalents of what you inject here? I mean, the whole point of this module is doing mails qweb style, why do you want to bend it back to jinja?

File: email_template_qweb/models/mail_template.py (hunk 1 above)
Comment (simahawk) [reply email_template_qweb/models/mail_template.py]: the comment is for the call to partner method only. I can cook a docstring tho

File: email_template_qweb/models/mail_template.py (hunk 2 above)
Comment (simahawk) [reply email_template_qweb/models/mail_template.py:83]: @hbrunn as I mentioned: it eases some copy/pasting from jinja templates. It makes easier for end users to customize templates. 

I could make them optional if you want or split to another module but seems over complicated for such a small thing.

Regarding "the whole point": I disagree, qweb is not only for using widgets and such. The key point about qweb is that you can reuse templates and its syntax is much more readable.

This is just a tiny layer of compatibility between them.

Comment (simahawk) [reply email_template_qweb/models/mail_template.py:83]: I could remove the formatters if we agree is "too much" :)

Comment (hbrunn) [reply email_template_qweb/models/mail_template.py:83]: no, keep it the way you have it. It doesn't hurt my qweb purism and helps others, so it's a good thing to do, thank you.

But https://codecov.io/gh/OCA/social/compare/b6dc1918b4aa8670a303edff99fcedeb3391250c...3efbe62cb6dea0b91e87be7391b750cc393a2553/diff#D2-44 needs to be addressed

**Comment by simahawk** [comment]
@hbrunn docstring + comments reworked. Coverage should be happy now as there's nothing to be tested on the core method call.

**Review by hbrunn — APPROVED** [review]

**Review by p-tombez — APPROVED** [review]

**Pull request merged**